src/workerd/server/workerd-api.c++

```diff
@@ -530,6 +516,10 @@ kj::Maybe<jsg::ModuleRegistry::ModuleInfo> WorkerdApi::tryCompileModule(jsg::Loc
       // Nothing to do. Handled in compileModules.
       return kj::none;
     }
+    default: {
```

Comment:
I think we should catch the `config::Worker::Module::OBSOLETE` specifically and throw rather than define default.
Not having defaults is generally good practice and makes extending the enum easier as compile errors direct you to more locations where the enum is used.